src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java

```diff
@@ -0,0 +1,255 @@
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    servlet.init(datastore);
+    gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+  }
+
+  @After
+  public void tearDown() {
+    localServiceHelper.tearDown();
+  }
+
+  @Test
+  public void doGet_getDataInDatastoreForShortVideo() throws ServletException, IOException {
+    putJsonInDatastore(SHORT_VIDEO_JSON, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doGet(request, response);
+    String actualJson = content.toString();
+    List<TranscriptLine> expectedArrayList = extractJsonAsArrayList(SHORT_VIDEO_JSON);
+    List<TranscriptLine> actualJsonArray = extractJsonAsArrayList(actualJson);
+
+    assertThat(actualJsonArray.size()).isEqualTo(expectedArrayList.size());
+  }
+
+  @Test
+  public void doPost_persistDataInDatastoreForShortVideo() throws ServletException, IOException {
+    when(request.getParameter(TranscriptServlet.PARAM_VIDEO_ID)).thenReturn(SHORT_VIDEO_ID);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_B);
+
+    servlet.doPost(request, response);
+    int actualQueryCount = countEntitiesInDatastore(LECTURE_ID_B);
+    int expectedQueryCount = (extractJsonAsArrayList(SHORT_VIDEO_JSON)).size();
+
+    assertThat(actualQueryCount).isEqualTo(expectedQueryCount);
+  }
+
+  @Test
+  public void doGet_doPost_StoreAndRetrieveShortVideo() throws ServletException, IOException {
+    when(request.getParameter(TranscriptServlet.PARAM_VIDEO_ID)).thenReturn(SHORT_VIDEO_ID);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doPost(request, response);
+    servlet.doGet(request, response);
+    String actualJson = content.toString();
+    List<TranscriptLine> expectedArrayList = extractJsonAsArrayList(SHORT_VIDEO_JSON);
+    List<TranscriptLine> actualJsonArray = extractJsonAsArrayList(actualJson);
+
+    assertThat(actualJsonArray).isEqualTo(expectedArrayList);
+  }
+
+  @Test
+  public void doGet_doPost_StoreAndRetrieveLongVideo() throws ServletException, IOException {
+    when(request.getParameter(TranscriptServlet.PARAM_VIDEO_ID)).thenReturn(LONG_VIDEO_ID);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doPost(request, response);
+    servlet.doGet(request, response);
+    String actualJson = content.toString();
+    List<TranscriptLine> expectedArrayList = extractJsonAsArrayList(LONG_VIDEO_JSON);
+    List<TranscriptLine> actualJsonArray = extractJsonAsArrayList(actualJson);
+
+    assertThat(actualJsonArray).isEqualTo(expectedArrayList);
+  }
+
+  @Test
+  public void doGet_getDataInDatastoreForLongVideo() throws ServletException, IOException {
```

Comment:
nit: for "getDataInDatastoreForLongVideo" should this be rephrased as "returnsLectureForLongVideoFromDatastore"

Reply:
done!